frontend/src/hooks/useApi.ts

```diff
@@ -0,0 +1,32 @@
+import { useCallback, useReducer } from "react";
+import axios from "axios";
+import { Status } from "../context";
+import reducer, { ReducerState } from "../context/reducer";
+
```

Comment:
The useApiRequest function is a custom hook (Some what advanced react) which abstracts logic away from presentation components. Hooks are meant to be reusable and in this case it's one that handles fetching api data.

It accepts and input as well as an options object for reusability.

Creates a new reducer (scoped to the component using useApiRequest) and returns an empty state and a callback called makerequest.

When the component using useApiRequest calls `makerequest()` this triggers axios to fetch what ever endpoint was specified. It also dispatched status + payloads as the promise resolves. Letting the component using the hook not need to worry about async timings.